### ISSUE TITLE
Twdaemon not correctly starting after upgrading to newer version

### ISSUE THREAD

**Issue by garrettrowell** [Tripwire/puppet-te_agent#1]
# Environment Details
* Operating System
  * CentOS Linux release 7.1.1503 (Core) 
* Puppet
  * puppet-agent.x86_64 5.0.1-1.el7
  * puppetserver.noarch 5.0.0-1.el7
* Tripwire
  * te_agent_8.3.7_en_linux_x86_64
  * te_agent_8.5.3_en_linux_x86_64
# Reproducing the Issue
## Notes
* The profile below uses the [puppet-staging (v2.2.0)](https://github.com/voxpupuli/puppet-staging/tree/v2.2.0) module to download and extract a tgz containing the desired version of the te_agent.
* The staging module extracts specified tgz into the */tmp* directory
  * For example the contents of `te_agent_8.3.7_en_linux_x86_64.tgz` gets extracted into */tmp/te_agent_8.3.7_en_linux_x86_64* which would then contain `te_agent.bin` 
* `te_agent::te_server_host` and `te_agent::te_services_passphrase` are being specified in hiera
## Steps to Reproduce
1. Use below profile to deploy te_agent 8.3.7 onto a fresh CentOS 7 machine
1. Use below profile to upgrade te_agent from version 8.3.7 to 8.5.3
## Profile Used
```puppet
class profiles::tripwire (
  String $te_agent_tar        = 'te_agent_8.3.7_en_linux_x86_64.tgz',
#  String $te_agent_tar        = 'te_agent_8.5.3_en_linux_x86_64.tgz',
  String $te_agent_tar_source = 'http://<internal-file-server>/files',
){

  staging::deploy { $te_agent_tar:
    source => "${te_agent_tar_source}/${te_agent_tar}",
    target => '/tmp/',
  }

  # determine extracted directory from te_agent_tar
  $_deploy_dir = regsubst($te_agent_tar, '^(.*)\..*$', '\1')

  # determine te_agent version from te_agent_tar
  $_deploy_version = regsubst($te_agent_tar, '^.*(\d\.\d\.\d).*$', '\1')

  # requires te_agent::te_server_host and te_agent::te_services_passphrase
  # to be set in hiera data
  class { 'te_agent':
    package_source => "/tmp/${_deploy_dir}/te_agent.bin",
    package_ensure => $_deploy_version,
    require        => Staging::Deploy[$te_agent_tar],
  }

}
```
# Expected Behaviour
Previously installed version (8.3.7) to be uninstalled and new version installed (8.5.3). After the upgrade twdaemon and twrtmd services get started correctly.
# Experienced Behaviour
* Clean install of version 8.3.7
  * twdaemon and twrtmd start correctly
* Version 8.3.7 Upgraded to 8.5.3
  * version 8.3.7 uninstalled and 8.5.3 installed correctly
  * twrtmd starts correctly
  * twdaemon does not start
### Clean install of 8.3.7
puppet agent -t
```bash
Notice: /Stage[main]/Profiles::Tripwire/Staging::Deploy[te_agent_8.3.7_en_linux_x86_64.tgz]/Staging::Extract[te_agent_8.3.7_en_linux_x86_64.tgz]/Exec[extract te_agent_8.3.7_en_linux_x86_64.tgz]/returns: executed successfully
Notice: /Stage[main]/Te_agent::Install/Package[Tripwire Enterprise Agent]/ensure: created
Notice: Augeas[agent.properties](provider=augeas): 
--- /usr/local/tripwire/te/agent/data/config/agent.properties	2017-08-24 13:09:35.304086589 -0400
+++ /usr/local/tripwire/te/agent/data/config/agent.properties.augnew	2017-08-24 13:09:38.425159244 -0400
@@ -14,3 +14,4 @@
 tw.server.host=<redacted>
 tw.server.port=9898
 tw.agent.generator.port=1169
+tw.local.port=9898

Notice: /Stage[main]/Te_agent::Config/Augeas[agent.properties]/returns: executed successfully
Info: Class[Te_agent::Config]: Scheduling refresh of Class[Te_agent::Service]
Info: Class[Te_agent::Service]: Scheduling refresh of Service[twdaemon]
Info: Class[Te_agent::Service]: Scheduling refresh of Service[twrtmd]
Notice: /Stage[main]/Te_agent::Service/Service[twdaemon]: Triggered 'refresh' from 1 event
Notice: /Stage[main]/Te_agent::Service/Service[twrtmd]/ensure: ensure changed 'stopped' to 'running'
Info: /Stage[main]/Te_agent::Service/Service[twrtmd]: Unscheduling refresh on Service[twrtmd]
```
puppet resource service twdaemon
```puppet
service { 'twdaemon':
  ensure => 'running',
  enable => 'false',
}
```
puppet resource service twrtmd
```puppet
service { 'twrtmd':
  ensure => 'running',
  enable => 'true',
}
```
systemctl status twdaemon
```bash
● twdaemon.service - (null)
   Loaded: loaded (/etc/rc.d/init.d/twdaemon; bad; vendor preset: disabled)
   Active: active (running) since Thu 2017-08-24 13:09:39 EDT; 7min ago
     Docs: man:systemd-sysv-generator(8)
   CGroup: /system.slice/twdaemon.service
           ├─10954 ./wrapper ./agent.conf wrapper.pidfile=../tmp/teagent.pid wrapper.daemonize=TRUE
           └─10955 /usr/local/tripwire/te/agent/jre/bin/java -da -server -Dtw.server=false -Djava.security.manager=com.tripwire.kernel.ext.security.TripwireSecurityManager -Djava.security.propert...

Aug 24 13:09:38 <hostname> systemd[1]: Starting (null)...
Aug 24 13:09:38 <hostname> twdaemon[10937]: Starting Tripwire Enterprise Agent...
Aug 24 13:09:39 <hostname> systemd[1]: Started (null).
```
systemctl status twrtmd
```bash
● twrtmd.service - LSB: Tripwire Real Time Monitoring Service
   Loaded: loaded (/etc/rc.d/init.d/twrtmd; bad; vendor preset: disabled)
   Active: active (running) since Thu 2017-08-24 13:09:39 EDT; 8min ago
     Docs: man:systemd-sysv-generator(8)
  Process: 10991 ExecStart=/etc/rc.d/init.d/twrtmd start (code=exited, status=0/SUCCESS)
 Main PID: 10907 (tesvc)
   CGroup: /system.slice/twrtmd.service
           ‣ 10907 /usr/local/tripwire/te/agent/bin/../sup/rtm/tesvc -port 1169

Aug 24 13:09:39 <hostname> systemd[1]: Starting LSB: Tripwire Real Time Monitoring Service...
Aug 24 13:09:39 <hostname> twrtmd[10991]: Real Time Monitoring is already started.
Aug 24 13:09:39 <hostname> systemd[1]: Started LSB: Tripwire Real Time Monitoring Service.
```
Contents of */usr/local/tripwire/te/agent/data/log/teagent.log*
```bash
STATUS | wrapper  | 2017/08/24 13:09:38 | --> Wrapper Started as Daemon
STATUS | wrapper  | 2017/08/24 13:09:38 | Java Service Wrapper Standard Edition 64-bit 3.5.25
STATUS | wrapper  | 2017/08/24 13:09:38 |   Copyright (C) 1999-2014 Tanuki Software, Ltd. All Rights Reserved.
STATUS | wrapper  | 2017/08/24 13:09:38 |     http://wrapper.tanukisoftware.com
STATUS | wrapper  | 2017/08/24 13:09:38 |   Licensed to Tripwire, Inc. for Tripwire Enterprise
STATUS | wrapper  | 2017/08/24 13:09:38 | 
STATUS | wrapper  | 2017/08/24 13:09:38 | Launching a JVM...
INFO   | jvm 1    | 2017/08/24 13:09:39 | WrapperManager: Initializing...
INFO   | jvm 1    | 2017/08/24 13:09:40 | TE Agent version: 8.3.7.b8.3.7.r20150126170702-46c11fb.b380
INFO   | jvm 1    | 2017/08/24 13:09:40 | Using Java version 1.7.0_72 from the path: /usr/local/tripwire/te/agent/jre
INFO   | jvm 1    | 2017/08/24 13:09:40 | Starting up on <hostname>:9898
INFO   | jvm 1    | 2017/08/24 13:09:40 | ProxySelector: No SOCKS proxy configured
INFO   | jvm 1    | 2017/08/24 13:09:40 | Bootstrap: station
INFO   | jvm 1    | 2017/08/24 13:09:40 | Connecting to services at <redacted>:9898
INFO   | jvm 1    | 2017/08/24 13:09:40 | RMI mode: 1
```
Contents of */usr/local/tripwire/te/agent/data/log/tesvc.log*
```bash
tesvc 2017/08/24 13:09:35.391 INFO:    CService::OnInitialize: logging initialized
tesvc 2017/08/24 13:09:35.391 INFO:    Settings: 
tesvc 2017/08/24 13:09:35.391 INFO:       debug-rule-trace: 0 
tesvc 2017/08/24 13:09:35.391 INFO:       disable-log: 0 
tesvc 2017/08/24 13:09:35.391 INFO:       allow-dump: 1 
tesvc 2017/08/24 13:09:35.391 INFO:       raise-priority: 1 
tesvc 2017/08/24 13:09:35.391 INFO:       logfile-path: /usr/local/tripwire/te/agent/sup/rtm/../../data/log/tesvc.log 
tesvc 2017/08/24 13:09:35.391 INFO:       log-limit-mb: 20 
tesvc 2017/08/24 13:09:35.391 INFO:       log-level: 1 
tesvc 2017/08/24 13:09:35.391 INFO:       port: 1169 
tesvc 2017/08/24 13:09:35.391 INFO:       dvrqueue: 10000 
tesvc 2017/08/24 13:09:35.391 INFO:       svcqueue: 50000 
tesvc 2017/08/24 13:09:35.391 INFO:       pidfile-path: /var/run/tesvc.pid 
tesvc 2017/08/24 13:09:35.391 INFO:       mem-report-secs: 0 
tesvc 2017/08/24 13:09:35.391 INFO:       max-msg-items: 50000 
tesvc 2017/08/24 13:09:35.391 INFO:    CService::OnInitialize: setting above normal priority
tesvc 2017/08/24 13:09:35.391 INFO:    CService::OnInitialize: setting priority done 
tesvc 2017/08/24 13:09:35.391 INFO:    CService::OnRunApp: cwd = '/usr/local/tripwire/te/agent/bin' 
tesvc 2017/08/24 13:09:35.391 INFO:    CService::OnRunApp: pid = 10907 
tesvc 2017/08/24 13:09:35.391 INFO:    CBaseWorkerThread(TASockSvrThrd): ctor done
tesvc 2017/08/24 13:09:35.391 INFO:    TASockSvrThrd::TASockSvrThrd
tesvc 2017/08/24 13:09:35.391 INFO:    CBaseWorkerThread(TASockSvrThrd): starting id:0x0000
tesvc 2017/08/24 13:09:35.391 INFO:    TASockSvrThrd::OnFirstRun() called...
tesvc 2017/08/24 13:09:35.891 INFO:    TASockSvrThrd:: state = connecting...
tesvc 2017/08/24 13:09:35.891 INFO:    TASockSvrThrd::DoWorkConnecting...
tesvc 2017/08/24 13:09:35.892 INFO:    TASockSvrThrd::DoConnecting: svr bind on port 1169
tesvc 2017/08/24 13:09:35.892 INFO:    TASockSvrThrd:: state = listening...
```
### Upgrading 8.3.7 to 8.5.3
puppet agent -t
```bash
Notice: /Stage[main]/Profiles::Tripwire/Staging::Deploy[te_agent_8.5.3_en_linux_x86_64.tgz]/Staging::Extract[te_agent_8.5.3_en_linux_x86_64.tgz]/Exec[extract te_agent_8.5.3_en_linux_x86_64.tgz]/returns: executed successfully
Notice: /Stage[main]/Te_agent::Install/Package[Tripwire Enterprise Agent]/ensure: ensure changed '8.3.7' to '8.5.3'
Notice: /Stage[main]/Te_agent::Service/Service[twrtmd]/ensure: ensure changed 'stopped' to 'running'
Info: /Stage[main]/Te_agent::Service/Service[twrtmd]: Unscheduling refresh on Service[twrtmd]
Info: Stage[main]: Unscheduling all events on Stage[main]
```
puppet resource service twdaemon
```puppet
service { 'twdaemon':
  ensure => 'stopped',
  enable => 'false',
}
```
puppet resource service twrtmd
```puppet
service { 'twrtmd':
  ensure => 'running',
  enable => 'true',
}
```
systemctl status twdaemon
```bash
● twdaemon.service - (null)
   Loaded: loaded (/etc/rc.d/init.d/twdaemon; bad; vendor preset: disabled)
   Active: active (exited) since Thu 2017-08-24 13:09:39 EDT; 21min ago
     Docs: man:systemd-sysv-generator(8)

Aug 24 13:09:38 <hostname> systemd[1]: Starting (null)...
Aug 24 13:09:38 <hostname> twdaemon[10937]: Starting Tripwire Enterprise Agent...
Aug 24 13:09:39 <hostname> systemd[1]: Started (null).
```
systemctl status twrtmd
```bash
● twrtmd.service - LSB: Tripwire Real Time Monitoring Service
   Loaded: loaded (/etc/rc.d/init.d/twrtmd; bad; vendor preset: disabled)
   Active: active (running) since Thu 2017-08-24 13:30:00 EDT; 2min 7s ago
     Docs: man:systemd-sysv-generator(8)
  Process: 13052 ExecStart=/etc/rc.d/init.d/twrtmd start (code=exited, status=0/SUCCESS)
 Main PID: 13019 (tesvc)
   CGroup: /system.slice/twrtmd.service
           ‣ 13019 /usr/local/tripwire/te/agent/bin/../sup/rtm/tesvc -port 1169

Aug 24 13:30:00 <hostname> systemd[1]: Starting LSB: Tripwire Real Time Monitoring Service...
Aug 24 13:30:00 <hostname> twrtmd[13052]: Real Time Monitoring is already started.
Aug 24 13:30:00 <hostname> systemd[1]: Started LSB: Tripwire Real Time Monitoring Service.
```
Contents of */usr/local/tripwire/te/agent/data/log/teagent.log*
```bash
STATUS | wrapper  | 2017/08/24 13:09:38 | --> Wrapper Started as Daemon
STATUS | wrapper  | 2017/08/24 13:09:38 | Java Service Wrapper Standard Edition 64-bit 3.5.25
STATUS | wrapper  | 2017/08/24 13:09:38 |   Copyright (C) 1999-2014 Tanuki Software, Ltd. All Rights Reserved.
STATUS | wrapper  | 2017/08/24 13:09:38 |     http://wrapper.tanukisoftware.com
STATUS | wrapper  | 2017/08/24 13:09:38 |   Licensed to Tripwire, Inc. for Tripwire Enterprise
STATUS | wrapper  | 2017/08/24 13:09:38 | 
STATUS | wrapper  | 2017/08/24 13:09:38 | Launching a JVM...
INFO   | jvm 1    | 2017/08/24 13:09:39 | WrapperManager: Initializing...
INFO   | jvm 1    | 2017/08/24 13:09:40 | TE Agent version: 8.3.7.b8.3.7.r20150126170702-46c11fb.b380
INFO   | jvm 1    | 2017/08/24 13:09:40 | Using Java version 1.7.0_72 from the path: /usr/local/tripwire/te/agent/jre
INFO   | jvm 1    | 2017/08/24 13:09:40 | Starting up on <hostname>:9898
INFO   | jvm 1    | 2017/08/24 13:09:40 | ProxySelector: No SOCKS proxy configured
INFO   | jvm 1    | 2017/08/24 13:09:40 | Bootstrap: station
INFO   | jvm 1    | 2017/08/24 13:09:40 | Connecting to services at <redacted>:9898
INFO   | jvm 1    | 2017/08/24 13:09:40 | RMI mode: 1
STATUS | wrapper  | 2017/08/24 13:29:45 | <-- Wrapper Stopped

```
Contents of */usr/local/tripwire/te/agent/data/log/tesvc.log*
```bash
Checking access to log dir: /usr/local/tripwire/te/agent/sup/rtm/../../data/log 
   succeeded 
Using logfile: /usr/local/tripwire/te/agent/sup/rtm/../../data/log/tesvc.log 
LoadConfigFile starting... 
LoadConfigFile trying file:'/etc/tesvc.conf' 
LoadConfigFile trying file:'/usr/local/tripwire/te/agent/sup/rtm/tesvc.conf' 
LoadConfigFile trying file:'/usr/local/tripwire/te/agent/sup/rtm/../../data/config/tesvc.conf' 
using pidfile /var/run/tesvc.pid 
using pidfile /var/run/tesvc.pid 
pid = 10907
logging to '/usr/local/tripwire/te/agent/sup/rtm/../../data/log/tesvc.log' 
tesvc 2017/08/24 13:29:57.673 INFO:    CService::OnInitialize: logging initialized
tesvc 2017/08/24 13:29:57.673 INFO:    Settings: 
tesvc 2017/08/24 13:29:57.673 INFO:       debug-rule-trace: 0 
tesvc 2017/08/24 13:29:57.673 INFO:       disable-log: 0 
tesvc 2017/08/24 13:29:57.673 INFO:       allow-dump: 1 
tesvc 2017/08/24 13:29:57.673 INFO:       raise-priority: 1 
tesvc 2017/08/24 13:29:57.673 INFO:       logfile-path: /usr/local/tripwire/te/agent/sup/rtm/../../data/log/tesvc.log 
tesvc 2017/08/24 13:29:57.673 INFO:       log-limit-mb: 20 
tesvc 2017/08/24 13:29:57.673 INFO:       log-level: 1 
tesvc 2017/08/24 13:29:57.673 INFO:       port: 1169 
tesvc 2017/08/24 13:29:57.673 INFO:       dvrqueue: 10000 
tesvc 2017/08/24 13:29:57.673 INFO:       svcqueue: 50000 
tesvc 2017/08/24 13:29:57.673 INFO:       pidfile-path: /var/run/tesvc.pid 
tesvc 2017/08/24 13:29:57.673 INFO:       mem-report-secs: 0 
tesvc 2017/08/24 13:29:57.673 INFO:       max-msg-items: 50000 
tesvc 2017/08/24 13:29:57.673 INFO:    CService::OnInitialize: setting above normal priority
tesvc 2017/08/24 13:29:57.673 INFO:    CService::OnInitialize: setting priority done 
tesvc 2017/08/24 13:29:57.673 INFO:    CService::OnRunApp: cwd = '/usr/local/tripwire/te/agent/bin' 
tesvc 2017/08/24 13:29:57.673 INFO:    CService::OnRunApp: pid = 13019 
tesvc 2017/08/24 13:29:57.673 INFO:    CBaseWorkerThread(TASockSvrThrd): ctor done
tesvc 2017/08/24 13:29:57.673 INFO:    TASockSvrThrd::TASockSvrThrd
tesvc 2017/08/24 13:29:57.673 INFO:    CBaseWorkerThread(TASockSvrThrd): starting id:0x0000
tesvc 2017/08/24 13:29:57.673 INFO:    TASockSvrThrd::OnFirstRun() called...
tesvc 2017/08/24 13:29:58.174 INFO:    TASockSvrThrd:: state = connecting...
tesvc 2017/08/24 13:29:58.174 INFO:    TASockSvrThrd::DoWorkConnecting...
tesvc 2017/08/24 13:29:58.174 INFO:    TASockSvrThrd::DoConnecting: svr bind on port 1169
tesvc 2017/08/24 13:29:58.174 INFO:    TASockSvrThrd:: state = listening...
```
# Proposed Solution
In *te_agent_bin.rb* if a previously installed version is detected, the old version gets uninstalled by `self.uninstall`. I found that after changing this to `self.purge` and going through the upgrade process outlined above, twdaemon starts as expected.
## Git Diff
```git
diff --git a/lib/puppet/provider/package/te_agent_bin.rb b/lib/puppet/provider/package/te_agent_bin.rb
index 441bc9b..265d7c7 100755
--- a/lib/puppet/provider/package/te_agent_bin.rb
+++ b/lib/puppet/provider/package/te_agent_bin.rb
@@ -29,7 +29,7 @@ Puppet::Type.type(:package).provide(:te_agent_bin, :parent => Puppet::Provider::
   def install
     if get_version
       # existing version must be uninstalled first
-      self.uninstall
+      self.purge
     end
     execute(['/bin/sh', resource[:source]] + install_options)
   end
```

**Comment by bespokebob** [comment]
Thanks for the detailed bug report.

I don't think that using `purge` is the best solution, as this will remove all other configuration files, logs, etc. on upgrade. We might need to "refresh" the service resources on upgrade instead.

**Comment by garrettrowell** [comment]
Admittedly while using ```purge``` worked in my use case, I could see how that would be undesirable elsewhere. Perhaps the following would be a better solution: 
#
**git diff**
``` bash
diff --git a/manifests/service.pp b/manifests/service.pp
index 79e408a..f904510 100755
--- a/manifests/service.pp
+++ b/manifests/service.pp
@@ -2,6 +2,12 @@
 # @api private
 class te_agent::service inherits te_agent {
 
+  if $te_agent::package_manage {
+    Service['twdaemon', 'twrtmd'] {
+      subscribe => Package['Tripwire Enterprise Agent'],
+    }
+  }
+
   if $te_agent::service_manage {
     service { 'twdaemon':
       ensure     => $te_agent::service_ensure,
```
This will subscribe each of the services to the installed package assuming that
*$te_agent::package_manage* == ```true```
#
Following the upgrade path outlined originally, after a clean install of 8.3.7 and upgrading to 8.5.3 the following was seen:

**puppet agent -t**
```bash
Notice: /Stage[main]/Profiles::Tripwire/Staging::Deploy[te_agent_8.5.3_en_linux_x86_64.tgz]/Staging::Extract[te_agent_8.5.3_en_linux_x86_64.tgz]/Exec[extract te_agent_8.5.3_en_linux_x86_64.tgz]/returns: executed successfully
Notice: /Stage[main]/Te_agent::Install/Package[Tripwire Enterprise Agent]/ensure: ensure changed '8.3.7' to '8.5.3'
Info: /Stage[main]/Te_agent::Install/Package[Tripwire Enterprise Agent]: Scheduling refresh of Service[twdaemon]
Info: /Stage[main]/Te_agent::Install/Package[Tripwire Enterprise Agent]: Scheduling refresh of Service[twrtmd]
Notice: /Stage[main]/Te_agent::Service/Service[twdaemon]: Triggered 'refresh' from 1 event
Notice: /Stage[main]/Te_agent::Service/Service[twrtmd]/ensure: ensure changed 'stopped' to 'running'
Info: /Stage[main]/Te_agent::Service/Service[twrtmd]: Unscheduling refresh on Service[twrtmd]
```

**puppet resource service twdaemon**
```puppet
service { 'twdaemon':
  ensure => 'running',
  enable => 'false',
}
```

**cat /usr/local/tripwire/te/agent/data/log/teagent.log**
```bash
STATUS | wrapper  | 2017/08/28 17:12:56 | Java Service Wrapper Standard Edition 64-bit 3.5.25
STATUS | wrapper  | 2017/08/28 17:12:56 |   Copyright (C) 1999-2014 Tanuki Software, Ltd. All Rights Reserved.
STATUS | wrapper  | 2017/08/28 17:12:56 |     http://wrapper.tanukisoftware.com
STATUS | wrapper  | 2017/08/28 17:12:56 |   Licensed to Tripwire, Inc. for Tripwire Enterprise
STATUS | wrapper  | 2017/08/28 17:12:56 | 
STATUS | wrapper  | 2017/08/28 17:12:56 | Launching a JVM...
INFO   | jvm 1    | 2017/08/28 17:12:57 | WrapperManager: Initializing...
INFO   | jvm 1    | 2017/08/28 17:12:57 | TE Agent version: 8.3.7.b8.3.7.r20150126170702-46c11fb.b380
INFO   | jvm 1    | 2017/08/28 17:12:57 | Using Java version 1.7.0_72 from the path: /usr/local/tripwire/te/agent/jre
INFO   | jvm 1    | 2017/08/28 17:12:57 | Starting up on <hostname>:9898
INFO   | jvm 1    | 2017/08/28 17:12:57 | ProxySelector: No SOCKS proxy configured
INFO   | jvm 1    | 2017/08/28 17:12:58 | Bootstrap: station
INFO   | jvm 1    | 2017/08/28 17:12:58 | Connecting to services at <redacted>:9898
INFO   | jvm 1    | 2017/08/28 17:12:58 | RMI mode: 1
STATUS | wrapper  | 2017/08/28 17:14:34 | TERM trapped.  Shutting down.
INFO   | jvm 1    | 2017/08/28 17:15:07 | ServerRegistry: Thread interrupted, could not get wrapped object of type interface com.tripwire.kernel.bootstrap.RmiCodebaseProvider
ERROR  | wrapper  | 2017/08/28 17:15:08 | Shutdown failed: Timed out waiting for signal from JVM.
ERROR  | wrapper  | 2017/08/28 17:15:09 | JVM did not exit on request, termination requested.
STATUS | wrapper  | 2017/08/28 17:15:09 | JVM received a signal SIGKILL (9).
STATUS | wrapper  | 2017/08/28 17:15:09 | JVM process is gone.
STATUS | wrapper  | 2017/08/28 17:15:09 | JVM exited after being requested to terminate.
STATUS | wrapper  | 2017/08/28 17:15:09 | <-- Wrapper Stopped
STATUS | wrapper  | 2017/08/28 17:15:25 | --> Wrapper Started as Daemon
STATUS | wrapper  | 2017/08/28 17:15:25 | Java Service Wrapper Standard Edition 64-bit 3.5.25
STATUS | wrapper  | 2017/08/28 17:15:25 |   Copyright (C) 1999-2014 Tanuki Software, Ltd. All Rights Reserved.
STATUS | wrapper  | 2017/08/28 17:15:25 |     http://wrapper.tanukisoftware.com
STATUS | wrapper  | 2017/08/28 17:15:25 |   Licensed to Tripwire, Inc. for Tripwire Enterprise
STATUS | wrapper  | 2017/08/28 17:15:25 | 
STATUS | wrapper  | 2017/08/28 17:15:26 | Launching a JVM...
INFO   | jvm 1    | 2017/08/28 17:15:27 | WrapperManager: Initializing...
INFO   | jvm 1    | 2017/08/28 17:15:27 | TE Agent version: 8.5.3.b8.5.3.r20170227192636-d0b5a65.b4
INFO   | jvm 1    | 2017/08/28 17:15:27 | Using Java version 1.8.0_121 from the path /usr/local/tripwire/te/agent/jre
INFO   | jvm 1    | 2017/08/28 17:15:27 | Initializing TE trust stores.
INFO   | jvm 1    | 2017/08/28 17:15:34 | Starting up on <hostname>:9898
INFO   | jvm 1    | 2017/08/28 17:15:34 | Bootstrap: station
INFO   | jvm 1    | 2017/08/28 17:15:34 | Connecting to services at <redacted>:9898
INFO   | jvm 1    | 2017/08/28 17:15:35 | RMISSLChannelFactory: Installing secure RMI Socket Factory
INFO   | jvm 1    | 2017/08/28 17:15:35 | RMISSLChannelFactory: JSSE is in non-FIPS mode
INFO   | jvm 1    | 2017/08/28 17:15:35 | Creating OU=TE RMI Host CA RSA, CN=<hostname> certificate.
```

**Comment by bespokebob** [comment]
This actually seems to be some strange interaction with the service init script and systemd. Puppet is running `systemctl is-active twdaemon` to determine the service status, and it comes back as "active", so it doesn't try to start the service. Using the init script directly returns the correct status.

Try setting `service_provider => 'redhat',` in your manifest and trying the upgrade again.

**Comment by garrettrowell** [comment]
:+1: just tested using your ```fix-service-provider``` branch and the upgrade worked correctly

**Comment by bespokebob** [comment]
Great, thanks!

**Comment by bespokebob** [comment]
Fixed in release v0.2.1
